src/main/java/io/github/nucleuspowered/nucleus/modules/environment/commands/WeatherCommand.java

```diff
@@ -88,6 +101,11 @@ public CommandResult executeCommand(CommandSource src, CommandContext args) thro
             oi =  i.isPresent() ? Optional.of((long)i.get()) : Optional.empty();
         }
 
+        if (oi.orElse(Long.MAX_VALUE) > eca.getNodeOrDefault().getMaximumWeatherTimespan() && !permissions.testSuffix(src, "exempt.length") && eca.getNodeOrDefault().getMaximumWeatherTimespan() != -1) {
+            src.sendMessage(plugin.getMessageProvider().getTextMessageWithFormat("command.weather.toolong", Util.getTimeStringFromSeconds(eca.getNodeOrDefault().getMaximumWeatherTimespan())));
```

Comment:
This isn't a success, this is an error condition. I've been moving to using `ReturnMessageException`s, so something like this is what you want.

`throw ReturnMessageException.fromKey("command.weather.toolong", Util.getTimeStringFromSeconds(eca.getNodeOrDefault().getMaximumWeatherTimespan()))`